dotCMS/src/integration-test/java/com/dotmarketing/portlets/contentlet/transform/ContentletTransformerTest.java

```diff
@@ -235,8 +304,329 @@ public void Test_Constant_fields_are_in_map() throws Exception {
+
+        TestCase(final BaseContentType baseContentType,
+                final List<Contentlet> contentlets) {
+            this.baseContentType = baseContentType;
+            this.contentlets = contentlets;
+        }
+    }
+
+    /**
+     * Given Scenario: We have the old and deprecated implementation vs the new one
+     * Expected Result: Both results must match
+     * @param testCase
+     */
+    @Test
+    @UseDataProvider("listTestCases")
+    public void Transformer_Backwards_Compatibility_Test(final TestCase testCase) {
+
+        final Set<String> privateInternalProperties = AbstractTransformStrategy.privateInternalProperties;
+
+        final List <Contentlet> list = testCase.contentlets;
+
+        if(list.isEmpty()){
+           Logger.warn(ContentletTransformerTest.class, String.format("Unable to get test samples of type `%s`",testCase.baseContentType.name()));
+           return;
+        }
+
+        final String baseTypeName = testCase.baseContentType.name();
+
+        final ContentletToMapTransformer contentletToMapTransformer = new ContentletToMapTransformer(list);
+
+        final List<Map<String, Object>> transformedList1 = contentletToMapTransformer.toMaps();
+
+        final DotContentletTransformer dotTransformer = new DotTransformerBuilder().defaultOptions().content(list).build();
+
+        final List<Map<String, Object>> transformedList2 = dotTransformer.toMaps();
+
+        assertEquals(transformedList1.size(), transformedList2.size());
+        for (int i = 0; i < list.size(); i++) {
+
+            final Contentlet original = list.get(i);
+            //We compare the transformation results are the same using the default options on the new Transformer
+            final Map<String, Object> sourceMap = transformedList1.get(i);
+            final Map<String, Object> copyMap = transformedList2.get(i);
+
+            assertTrue(String.format(" baseType `%s` should have same (or more) number of properties " ,baseTypeName),copyMap.size() >= sourceMap.size());
+
+            for (final String propertyName : sourceMap.keySet()) {
+
+                final Object object1 = sourceMap.get(propertyName);
+                final Object object2 = copyMap.get(propertyName);
+
+                if(null != object1 && object2 == null &&  privateInternalProperties.contains(propertyName)){
+                   //We're looking at private property that exists in the source but was removed on the copy. which is fine.
+                   continue;
+                }
+
+                if(object1 instanceof File){
+                    //Binaries are now formatted to their /dA/ path form.
+                    final File conBinary = (File)object1;
+                    final String dAPath = "/dA/%s/%s/%s";
+                    final String binaryPath = String.format(dAPath, sourceMap.get("identifier"),propertyName,conBinary.getName());
+                    assertEquals(String.format(" contentType:`%s` , id: `%s` ,  key: `%s` ",
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal " contentType:`%s` , id: `%s` ,  key: `%s` " appears 4 times in this file; the first occurrence is on line 400](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5596402)